be/feat: github actions 배포 설정

dlwldnjs1009 opened the pull request:
## 변경 사항
github actions .yml 파일 설정했습니


## 테스트 결과 (테스트를 했으면)
테스트는 push가 되어봐야 테스트를 할 수 있어서 테스트를 아직 못해봤습니다.


## +α Checklist
- [ ] 자바 코드 컨벤션을 지키면서 프로그래밍했는가?
- [ ] 한 메서드에 오직 한 단계의 들여쓰기(indent)만 허용했는가?
- [ ] else 예약어를 쓰지 않았는가?
- [ ] 모든 원시값과 문자열을 포장했는가?
- [ ] 콜렉션에 대해 일급 콜렉션을 적용했는가?
- [ ] 3개 이상의 인스턴스 변수를 가진 클래스를 구현하지 않았는가? (가능하면 인스턴스 변수의 수를 줄이기 위해 노력한다.)
- [ ] getter/setter 없이 구현했는가? (단, DTO는 허용한다.)
- [ ] 메소드의 인자 수를 제한했는가? (4개 이상의 인자는 허용하지 않는다. 3개도 가능하면 줄이기 위해 노력해 본다.)
- [ ] 코드 한 줄에 점(.)을 하나만 허용했는가?
- [ ] 메소드가 한가지 일만 담당하도록 구현했는가?
- [ ] 클래스를 작게 유지하기 위해 노력했는가?

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced an automated deployment pipeline that streamlines backend updates and production releases.
  - Added a dedicated process for building production-ready containers.

- **Chores**
  - Streamlined the build process by disabling legacy tasks.

- **Refactor**
  - Expanded the range of trusted domains for cross-origin requests to improve integration support.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

dlwldnjs1009 committed:
be/feat: github actions 배포 설정

lsh1215 reviewed (dismissed):
확인 완료했습니다. 추가적으로 머지되면 확인해서 알려주세요

dlwldnjs1009 committed:
fix: backend-cd.yml

이벤트 트리거:
on: 아래에 push 이벤트는 develop 브랜치에, 그리고 pull_request 이벤트는 main과 develop 브랜치에 대해 트리거되도록 수정

기본 작업 디렉토리:
jobs.deploy.defaults.run.working-directory를 backend로 지정해 모든 run 단계가 backend 디렉토리에서 실행되도록 수정

lsh1215 reviewed (approved):
goooood!
